gensim/models/word2vec.py

```diff
@@ -941,20 +948,26 @@ def _do_train_job(self, sentences, alpha, inits):
         work, neu1 = inits
         tally = 0
         if self.sg:
-            tally += train_batch_sg(self, sentences, alpha, work, self.compute_loss)
+            tally += train_batch_sg(
+                self, sentences, alpha, work,
+                self.compute_loss, self.reduced_windows,
+            )
         else:
-            tally += train_batch_cbow(self, sentences, alpha, work, neu1, self.compute_loss)
+            tally += train_batch_cbow(
+                self, sentences, alpha, work, neu1,
+                self.compute_loss, self.reduced_windows,
+            )
         return tally, self._raw_word_count(sentences)
 
     def _clear_post_train(self):
         """Clear any cached values that training may have invalidated."""
         self.wv.norms = None
 
     def train(
-            self, corpus_iterable=None, corpus_file=None, total_examples=None, total_words=None,
-            epochs=None, start_alpha=None, end_alpha=None, word_count=0,
-            queue_factor=2, report_delay=1.0, compute_loss=False, callbacks=(),
-            **kwargs,
+            self, corpus_iterable=None, corpus_file=None, total_examples=None,
+            total_words=None, epochs=None, start_alpha=None, end_alpha=None,
+            word_count=0, queue_factor=2, report_delay=1.0, compute_loss=False,
+            reduced_windows=True, callbacks=(), **kwargs,
```

Comment:
Per above: this parameter shouldn't need to be here. (Further: if it were here, it shouldn't override a possible user-chosen value of `False` from elsewhere.)

Reply:
I have mixed feeling, influenced by the fact that `compute_loss` is an example of a parameter overridden here in spite of its being also declared at instantiation. So I think it might be useful to leave the parameter open for overriding here - but if you disagree I will of course take it off.
At the very least I agree with the fact that it should not override the preset value if left to a default `None` value - a behaviour corrected in the upcoming commit.

Reply:
The current haphazard attempts at computing running loss are a big mess - see for example #2617 & other issues referenced therein. So I'd not use this `compute_loss` parameter as a model of good practice. Notably, also, loss-computation *might* involve an extra overhead people would only sometimes want to pay across multiple `train()` callss, and shouldn't change algorithmic behavior at all. But for enabling/disabling variable-windows, that *does* change the algorithm's operation, and I can't really imagine a case where someone would want to try it both ways in the same model, on subsequent `.train()` calls. So having this novel algorithmic parameter here – but not many others! –  doesn't fit.

Reply:
Thank you for the detailed argument; this makes perfect sense and the last commit on Friday removes the parameter from `train`.